aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/UpdateHandler.java

```diff
@@ -263,6 +299,33 @@ private boolean shouldUnsetMaxAllocatedStorage(final ResourceHandlerRequest<Reso
                 request.getDesiredResourceState().getMaxAllocatedStorage() == null;
     }
 
+
+    private boolean isAllocatedStorageIncrease(
+            final ResourceHandlerRequest<ResourceModel> request
+    ) {
+        return BooleanUtils.isNotTrue(request.getRollback()) &&
+                request.getPreviousResourceState() != null &&
+                Translator.getAllocatedStorage(request.getDesiredResourceState()) > Translator.getAllocatedStorage(request.getPreviousResourceState());
+    }
+
+    private boolean isInstanceInStorageFullState(
+            final ResourceHandlerRequest<ResourceModel> request,
+            final ProxyClient<RdsClient> rdsProxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress
+    ) {
+        // need to store this in the context to prevent premature exit from storage-full-handle
+        if (progress.getCallbackContext().isStorageFullInProgress()) {
+            return true;
+        }
+
+        try {
+            final DBInstance instance = fetchDBInstance(rdsProxyClient, request.getDesiredResourceState());
+            return Objects.equals(instance.dbInstanceStatus(), STORAGE_FULL_STATUS);
+        } catch (Exception ex) {
+            return false;
+        }
```

Comment:
Let's bubble this up, I guess. Such an non-selective catch doesn't sound reliable enough to return `false`.